File: tests/lang-test/go/utils/strings.go
package utils

import "strings"

// ripex-lang-test: Go string utils — closures, slices.
func Greet(name string) string {
	return "Hello, " + name + "!"
}

func MaskEmail(email string) string {
	parts := strings.Split(email, "@")
	return parts[0][:1] + "***@" + parts[1]
}

func FilterPositive(xs []int) []int {
	var out []int
	for _, x := range xs {
		if x > 0 {
			out = append(out, x)
		}
	}
	return out
}

File: tests/lang-test/rust/src/utils/strings.rs
//! ripex-lang-test: Rust string utils — closures, lifetimes.
pub fn greet(name: &str) -> String {
    format!("Hello, {}!", name)
}

pub fn mask_email(email: &str) -> String {
    let parts: Vec<&str> = email.split('@').collect();
    format!("{}***@{}", &parts[0][..1], parts[1])
}

pub fn filter_positive<'a>(xs: &'a [i32]) -> Vec<i32> {
    xs.iter().filter(|&&x| x > 0).copied().collect()
}
